lib/query-interface.js

```diff
@@ -462,6 +462,51 @@ class QueryInterface {
     return this.sequelize.query(sql, options);
   }
 
+  addConstraint(tableName, attributes, options, rawTablename) {
+    if (!options.type) {
+      throw new Error('Constraint type must be specified through options.type');
+    }
+    if (!Array.isArray(attributes)) {
+      rawTablename = options;
+      options = attributes;
+      attributes = options.fields;
+    }
+    // testhint argsConform.end
+
+    if (!rawTablename) {
+      // Map for backwards compat
+      rawTablename = tableName;
+    }
+
+    options = Utils.cloneDeep(options);
+    options.fields = attributes;
+    if (this.sequelize.dialect.name === 'sqlite') {
+      return SQLiteQueryInterface.addConstraint.call(this, tableName, options, rawTablename);
```

Comment:
@felixfbecker Only QueryGenerators are inherited. [`removeColumn`](https://github.com/sequelize/sequelize/blob/master/lib/query-interface.js#L333-L348) also uses the same pattern.